lib/command.js

```diff
@@ -479,13 +479,17 @@ class Command extends GetterSetter {
         this._program
       ));
     }
-    const actionResults = this._action.apply(this, [args, options, this._logger]);
-    const response = Promise.resolve(actionResults);
-    return response
-      .catch(err => {
-        err = err instanceof Error ? err : new Error(err);
-        throw this._program.fatalError(err);
-      })
+    try {
+      const actionResults = this._action.apply(this, [args, options, this._logger]);
+      return Promise.resolve(actionResults)
+        .catch(err => {
+          err = err instanceof Error ? err : new Error(err);
+          throw this._program.fatalError(err);
+        })
+    } catch (err) {
+      const e = err instanceof Error ? err : new Error(err);
+      throw this._program.fatalError(e);
+    }
```

Comment:
I think that we can simplify this by:

``` js
      return Promise.resolve(this._action.apply(this, [args, options, this._logger]))
        .catch(err => {
          err = err instanceof Error ? err : new Error(err);
          throw this._program.fatalError(err);
        })

```

this way, .catch() should be called even if some error is thrown in this._action()
Not tested though!

Reply:
Unfortunately, this won't work:

`this._action.apply()` is executed before the promise handling.  If `this._action.apply()` immediately throws an exception, it won't be handled by the `.catch()` function.

Reply:
Right, maybe more something like:

```js
return new Promise((resolve) => {
    resolve(this._action.apply(this, [args, options, this._logger]))
}).catch(err => {
    err = err instanceof Error ? err : new Error(err);
    throw this._program.fatalError(err);
})
```

Reply:
Throwing exceptions from inside the `.catch()` function doesn't result in code going down the correct path.